wodles/aws/buckets_s3/waf.py

```diff
@@ -97,18 +97,21 @@ def get_service_prefix(self, account_id):
         
     def get_full_prefix(self, account_id, account_region, acl_name=None):
         if self.type == WAF_NATIVE:
-            bucket_path = self.client.list_objects_v2(Bucket=self.bucket)
-            if 'Contents' in bucket_path:
-                contents = bucket_path.get('Contents')
-                # Retrieves the last key of the contents
-                log_key = contents[-1]['Key']
-                parts = log_key.split("/")
-                try:
-                    acl_name = parts[parts.index("WAFLogs") + 2]
-                except (ValueError, IndexError):
-                    raise ValueError("Could not find ACL name in the object.")
-                
-                return AWSLogsBucket.get_full_prefix(self, account_id, account_region, acl_name)
+            if self.waf_acls:
+                return AWSLogsBucket.get_full_prefix(self, account_id, account_region, self.waf_acls)
+            else: 
+                bucket_path = self.client.list_objects_v2(Bucket=self.bucket)
+                if 'Contents' in bucket_path:
+                    contents = bucket_path.get('Contents')
+                    # Retrieves the last key of the contents
+                    log_key = contents[-1]['Key']
+                    parts = log_key.split("/")
+                    try:
+                        acl_name = parts[parts.index("WAFLogs") + 2]
+                    except (ValueError, IndexError):
+                        raise ValueError("Could not find ACL name in the object.")
+                    
+                    return AWSLogsBucket.get_full_prefix(self, account_id, account_region, acl_name)
```

Comment:
No need to use `else` here, since the `if` condition exits the function

```suggestion
            bucket_path = self.client.list_objects_v2(Bucket=self.bucket)
            if 'Contents' in bucket_path:
                contents = bucket_path.get('Contents')
                # Retrieves the last key of the contents
                log_key = contents[-1]['Key']
                parts = log_key.split("/")
                try:
                    acl_name = parts[parts.index("WAFLogs") + 2]
                except (ValueError, IndexError):
                    raise ValueError("Could not find ACL name in the object.")
                    
                return AWSLogsBucket.get_full_prefix(self, account_id, account_region, acl_name)
```